### PR TITLE
Refactored server.wsRouter to avoid huge switch statement

### PR DESCRIPTION
**Pull request by carlosarismendi**

### CODE REVIEW

File: backend/server/wsrouter.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"fmt"
+	"log"
+)
+
+type WsHandler func([]byte) error
+
+type WsRouter struct {
+	handlers map[string]WsHandler
+}
```
Comment (Visn0) [inline backend/server/wsrouter.go]: Need mutex.

File: backend/server/server.go [hunk 2]
```diff
@@ -197,21 +197,26 @@ func (s *Server) wsRouter(room *domain.Room, c domain.ConnectionRepository, isHo
 		fmt.Println("Calculating valid moves for white player")
 	}
 
+	wsRouter := NewWsRouter()
+	wsRouter.Add("request-moves", infrastructure.NewGetValidMovesWsController(c, room.Game).Invoke)
+	wsRouter.Add("move-piece", infrastructure.NewMovePieceWsController(player, enemy, c, cEnemy, room.Game).Invoke)
+	wsRouter.Add("get-timers", infrastructure.NewGetTimersWsController(c, player, enemy).Invoke)
+	wsRouter.Add("abandon", infrastructure.NewAbandonWsController(cEnemy).Invoke)
+	wsRouter.Add("request-draw", infrastructure.NewRequestDrawWsController(cEnemy).Invoke)
+	wsRouter.Add("response-draw", infrastructure.NewResponseDrawWsController(cEnemy).Invoke)
```
Comment (Visn0) [inline backend/server/server.go]: Better to put in the constructor, more efficient (?)

File: backend/server/wsrouter.go (hunk 1 above)
Comment (carlosarismendi) [reply backend/server/wsrouter.go]: Why? Each user will have its own router and therefore, there will only be a routine accessing a router. No need of mutex.

File: backend/server/server.go (hunk 2 above)
Comment (carlosarismendi) [reply backend/server/server.go]: Changed.

**Pull request merged**